src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -311,71 +377,52 @@ void ZoneWindow::InitializeId(PCWSTR deviceId, PCWSTR virtualDesktopId) noexcept
 
 void ZoneWindow::LoadSettings() noexcept
 {
-    wchar_t activeZoneSetId[256];
-    RegistryHelpers::GetString(m_uniqueId, L"ActiveZoneSetId", activeZoneSetId, sizeof(activeZoneSetId));
-    CLSIDFromString(activeZoneSetId, &m_activeZoneSetId);
+    auto& deviceInfoMap = JSONHelpers::FancyZonesDataInstance().GetDeviceInfoMap();
+    if (!deviceInfoMap.contains(m_uniqueId))
+    {
+        JSONHelpers::FancyZonesDataInstance().MigrateDeviceInfoFromRegistry(m_uniqueId);
+    }
+  
+    JSONHelpers::FancyZonesDataInstance().GetDeviceInfoFromTmpFile(m_uniqueId, m_activeZoneSetPath);
+    const WCHAR* activeZoneSetStr = deviceInfoMap[m_uniqueId].activeZoneSet.uuid.c_str();
+
+    JSONHelpers::FancyZonesDataInstance().GetDeletedCustomZoneSetsFromTmpFile(m_customZoneSetsPath);
 }
 
 void ZoneWindow::InitializeZoneSets(MONITORINFO const& mi) noexcept
 {
-    LoadZoneSetsFromRegistry();
+    CalculateZoneSet();
 
     if (!m_activeZoneSet)
     {
         ChooseDefaultActiveZoneSet();
     }
 }
 
-void ZoneWindow::LoadZoneSetsFromRegistry() noexcept
-{
-    wil::unique_hkey key{RegistryHelpers::OpenKey(m_workArea)};
-    if (!key)
-    {
-        return;
-    }
-
-    ZoneSetPersistedData data{};
-    DWORD dataSize = sizeof(data);
-    wchar_t value[256]{};
-    DWORD valueLength = ARRAYSIZE(value);
-    DWORD i = 0;
-    while (RegEnumValueW(key.get(), i++, value, &valueLength, nullptr, nullptr, reinterpret_cast<BYTE*>(&data), &dataSize) == ERROR_SUCCESS)
-    {
-        if (data.Version == VERSION_PERSISTEDDATA)
-        {
-            GUID zoneSetId;
-            if (SUCCEEDED_LOG(CLSIDFromString(value, &zoneSetId)))
-            {
-                auto zoneSet = MakeZoneSet(ZoneSetConfig(
-                    zoneSetId,
-                    data.LayoutId,
-                    m_monitor,
-                    m_workArea));
-
-                if (zoneSet)
-                {
-                    for (UINT j = 0; j < data.ZoneCount; j++)
-                    {
-                        zoneSet->AddZone(MakeZone(data.Zones[j]));
-                    }
-
-                    if (zoneSetId == m_activeZoneSetId)
-                    {
-                        UpdateActiveZoneSet(zoneSet.get());
-                    }
-
-                    m_zoneSets.emplace_back(std::move(zoneSet));
-                }
-            }
-        }
-        else
-        {
-            // Migrate from older settings format
-        }
-
-        valueLength = ARRAYSIZE(value);
-        dataSize = sizeof(data);
-    }
+void ZoneWindow::CalculateZoneSet() noexcept
+{
+  const auto& deviceInfoMap = JSONHelpers::FancyZonesDataInstance().GetDeviceInfoMap();
```

Comment:
no need for using reference in functions returning by value. RVO should eliminate unnecessary copy.